File: AnalyticsKit/AnalyticsKit/AnalyticsKitAppDelegate.m
//
//  AnalyticsKitAppDelegate.m
//  AnalyticsKit
//
//  Created by Christopher Pickslay on 10/23/13.
//  Copyright (c) 2013 Two Bit Labs. All rights reserved.
//

#import "AnalyticsKitAppDelegate.h"

@implementation AnalyticsKitAppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    [AnalyticsKit initializeProviders:@[[AnalyticsKitDebugProvider new]]];
    [AnalyticsKit logEvent:@"App started"];
    return YES;
}
							
- (void)applicationWillEnterForeground:(UIApplication *)application {
    [AnalyticsKit applicationWillEnterForeground];
}

- (void)applicationDidEnterBackground:(UIApplication *)application {
    [AnalyticsKit applicationDidEnterBackground];
}

- (void)applicationWillTerminate:(UIApplication *)application {
    [AnalyticsKit applicationWillTerminate];
}


@end

File: AnalyticsKit/AnalyticsKit/AnalyticsKitAppDelegate.swift
import UIKit

@UIApplicationMain
class AnalyticsKitAppDelegate: UIResponder, UIApplicationDelegate {
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey : Any]? = nil) -> Bool {
        AnalyticsKit.initializeProviders([AnalyticsKitDebugProvider()])
        AnalyticsKit.logEvent("App started")
        return true
    }

    func applicationWillEnterForeground(_ application: UIApplication) {
        AnalyticsKit.applicationWillEnterForeground()
    }

    func applicationDidEnterBackground(_ application: UIApplication) {
        AnalyticsKit.applicationDidEnterBackground()
    }

    func applicationWillTerminate(_ application: UIApplication) {
        AnalyticsKit.applicationWillTerminate()
    }
}
